### PR TITLE
Updated image features

### PR DESCRIPTION
**Pull request by ShreyaR**

### CODE REVIEW

File: docs/configuration/features/image_features.md [hunk 1]
```diff
@@ -1,76 +1,340 @@
...
   width: 128
+  num_channels: 3
   resize_method: interpolate
   scaling: pixel_normalization
+  in_memory: true
+  num_processes: 4
 ```
 
-## Image Input Features and Encoders
+# Image Input Features and Encoders
```
Comment (dantreiman) [inline docs/configuration/features/image_features.md:142]: This is h2 in the other features docs, lets keep it that way for now and make a global style change later.

File: docs/configuration/features/image_features.md [hunk 2]
```diff
@@ -1,76 +1,340 @@
...
-- `fc_norm` (default `null`): if a `norm` is not already specified in `fc_layers` this is the default `norm` that will be used for each layer. It indicates the norm of the output and it can be `null`, `batch` or `layer`.
-- `fc_norm_params` (default `null`): parameters used if `norm` is either `batch` or `layer`.  For information on parameters used with `batch` see [Tensorflow's documentation on batch normalization](https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization) or for `layer` see [Tensorflow's documentation on layer normalization](https://www.tensorflow.org/api_docs/python/tf/keras/layers/LayerNormalization).
-- `fc_activation` (default `relu`): if an `activation` is not already specified in `fc_layers` this is the default `activation` that will be used for each layer. It indicates the activation function applied to the output.
-- `fc_dropout` (default `0`): dropout rate
+## Convolutional Stack Encoder (`stacked_cnn`)
```
Comment (dantreiman) [inline docs/configuration/features/image_features.md:146]: h3 for consistency

File: docs/configuration/features/image_features.md [hunk 3]
```diff
@@ -1,76 +1,340 @@
...
+
+### `conv_bias_initializer`
+
+ (default `'zeros'`):  initializer for the bias vector. Options are: `constant`, `identity`, `zeros`, `ones`, `orthogonal`, `normal`, `uniform`, `truncated_normal`, `variance_scaling`, `glorot_normal`, `glorot_uniform`, `xavier_normal`, `xavier_uniform`, `he_normal`, `he_uniform`, `lecun_normal`, `lecun_uniform`. Alternatively it is possible to specify a dictionary with a key `type` that identifies the type of initializer and other keys for its parameters, e.g. `{type: normal, mean: 0, stddev: 0}`. To know the parameters of each initializer, please refer to [TensorFlow's documentation](https://www.tensorflow.org/api_docs/python/tf/keras/initializers).
+
+### `weights_regularizer`
```
Comment (dantreiman) [inline docs/configuration/features/image_features.md:247]: The *_regularizer parameters (`weights_regularizer`, `bias_regularizer`...) for layers are gone as of 0.5 - remove

**Review by dantreiman — APPROVED** [review]
Look good main feedback is regarding the use of heading levels, which also relates to why precommit is failing.  Recommend changing headings to match the other feature docs for now.  (though I also agree we could improve things with a style update for readability, lets do that in a later PR)

File: docs/configuration/features/image_features.md [hunk 4]
```diff
@@ -1,76 +1,340 @@
-## Image Features Preprocessing
+Input image features are transformed into a float valued tensors of size `N x C x H x W` (where `N` is the size of the dataset, `C` is the number of channels, and `H x W` is the height and width of the image (can be specified by the user). These tensors are added to HDF5 with a key that reflects the name of column in the dataset.
+
+The column name is added to the JSON file, with an associated dictionary containing preprocessing information about the sizes of the resizing.
+
+# Supported Image Formats
+
+The number of channels in the image is determined by the image format. The following table lists the supported image formats and the number of channels.
+
+| Format      | Number of channels |
```
[the remark this answers is not in the document]
Comment (ShreyaR) [reply docs/configuration/features/image_features.md:9]: Thanks!

File: docs/configuration/features/image_features.md [hunk 5]
```diff
@@ -1,76 +1,340 @@
...
-
-Depending on the application, it is preferrable not to exceed a size of `256 x 256`, as bigger sizes will, in most cases, not provide much advantage in terms of performance, while they will considerably slow down training and inference and also make both forward and backward passes consume considerably more memory, leading to memory overflows on machines with limited amounts of RAM or on GPUs with limited amounts of VRAM.
+
+!!! note
+    Images passed to an image encoder are expected to have the same size. If images are different sizes, by default they will be resized to the dimenions of the first image in the dataset. Optionally, a `resize_method`, together with a target `width` and `height`, can be specified in the feature preprocessing parameters, in which case all images will be resized to the specified target size.
+
+### `missing_value_strategy`
```
[the remark this answers is not in the document]
Comment (ShreyaR) [reply docs/configuration/features/image_features.md:23]: @dantreiman following the discussion in the ML standup, I'll keep the formatting changes for now so as to avoid duplication of effort.

File: docs/configuration/features/image_features.md (hunk 1 above)
Comment (ShreyaR) [reply docs/configuration/features/image_features.md:142]: Same as above, keeping the headers this way for now to avoid duplication of effort.

File: docs/configuration/features/image_features.md (hunk 2 above)
Comment (ShreyaR) [reply docs/configuration/features/image_features.md:146]: Same as above, keeping the headers this way for now to avoid duplication of effort.

File: docs/configuration/features/image_features.md (hunk 3 above)
Comment (ShreyaR) [reply docs/configuration/features/image_features.md]: @dantreiman thanks for pointing that out, I've removed the references to conv regularizers.

I did a quick skim through the code and it looks like that there are still fully connected weight regularizers, so I've kept those in for now.

**Comment by ShreyaR** [comment]
@dantreiman responded to your comments, and removed references to conv regularizers!
